Source/Boxed.DotnetNewTest/DotnetNew.cs

```diff
@@ -86,17 +78,63 @@ await ProcessExtensions
         /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
         public static async Task ReinitialiseAsync(TimeSpan? timeout = null, bool showShellWindow = false)
         {
-            using (var cancellationTokenSource = new CancellationTokenSource(timeout ?? ConfigurationService.DefaultTimeout))
+            await RunDotnetCommandAsync($"new --debug:reinit", timeout, showShellWindow).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Uninstalls a template from the specified source.
+        /// </summary>
+        /// <typeparam name="T">A type from the assembly used to find the directory path of the project to install.</typeparam>
+        /// <param name="fileName">Name of the file.</param>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        public static Task UninstallAsync<T>(string fileName) =>
+            UninstallAsync(typeof(T).GetTypeInfo().Assembly, fileName);
+
+        /// <summary>
+        /// Uninstalls a template from the specified source.
+        /// </summary>
+        /// <param name="assembly">The assembly used to find the directory path of the project to install.</param>
+        /// <param name="fileName">Name of the file.</param>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        /// <exception cref="ArgumentNullException">The provided assembly was null.</exception>
+        /// <exception cref="FileNotFoundException">A file with the specified file name was not found.</exception>
+        public static Task UninstallAsync(Assembly assembly, string fileName)
+        {
+            if (assembly is null)
             {
-                await ProcessExtensions
-                    .StartAsync(
-                        DirectoryExtensions.GetCurrentDirectory(),
-                        "dotnet",
-                        $"new --debug:reinit",
-                        showShellWindow,
-                        cancellationTokenSource.Token)
-                    .ConfigureAwait(false);
+                throw new ArgumentNullException(nameof(assembly));
+            }
+
+            if (fileName is null)
+            {
+                throw new ArgumentNullException(nameof(fileName));
+            }
+
+            var projectFilePath = Path.GetDirectoryName(GetFilePath(assembly, fileName));
```

Comment:
We can probably pull this into a function as the same code to find the projectFilePath is used above for the Install.